fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -1194,6 +1194,38 @@ public CommandProcessingResult makeInterestPaymentWaiver(final JsonCommand comma
                 .build();
     }
 
+    @Transactional
+    @Override
+    public CommandProcessingResult makeAccrualActivity(JsonCommand command) {
```

Comment:
Not needed this way. Only job and business step will call. The incoming parameter will be Loan. And this service must calculate the balances based on the accruals on the actual period and create the transaction.